db/migrate/20250427103023_add_not_null_constraint_to_year_on_years.rb

```diff
@@ -0,0 +1,5 @@
+class AddNotNullConstraintToYearOnYears < ActiveRecord::Migration[7.1]
+  def change
+    change_column_null :years, :year, false
```

Comment:
This is assuming that we do not have existing rows with null values or that we are concerned with Postgresql blocking reads and write, while it checks existing rows for `NULL` values.